### PR TITLE
dbeaver/pro#2484 Google BQ: show the bytes processed or bytes billed

### PR DESCRIPTION
**Pull request by serjiokov**
Facades (factory, result set, session)for BigQuery

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.bigquery/src/org/jkiss/dbeaver/ext/bigquery/model/data/BigQueryTrace.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    private Map<String, String> resultDetails;
+    private final ByteNumberFormat byteFormater = new ByteNumberFormat();
+
+    public BigQueryTrace(@NotNull BigQueryResultSet resultSet) {
+        resultDetails = new LinkedHashMap<>();
+        try {
+            Object mResultSet = BeanUtils.getInheritedFieldValue(resultSet.getOriginal(), "m_resultSet"); //$NON-NLS-1$
+            Object mDataHandler = BeanUtils.getInheritedFieldValue(mResultSet, "m_dataHandler"); //$NON-NLS-1$
+            Object mBqResults = BeanUtils.getInheritedFieldValue(mDataHandler, "m_bqResults"); //$NON-NLS-1$
+            Object mQueryResponse = BeanUtils.getInheritedFieldValue(mBqResults, "m_queryResponse"); //$NON-NLS-1$
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.bigquery/src/org/jkiss/dbeaver/ext/bigquery/model/data/BigQueryTrace.java]: This feels really unsafe. Are there no public getters available so we can at least assume these wouldn't change in the next minor release?

File: plugins/org.jkiss.dbeaver.ext.bigquery/src/org/jkiss/dbeaver/ext/bigquery/model/data/BigQueryTrace.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import org.jkiss.dbeaver.model.exec.trace.DBCTrace;
+import org.jkiss.utils.BeanUtils;
+import org.jkiss.utils.ByteNumberFormat;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class BigQueryTrace implements DBCTrace {
+    private static final Log log = Log.getLog(BigQueryTrace.class);
+    private Map<String, String> resultDetails;
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.bigquery/src/org/jkiss/dbeaver/ext/bigquery/model/data/BigQueryTrace.java]: See my comment in the other repo.

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

**Comment by serjiokov** [comment]
image

**Comment by serjiokov** [comment]
image

File: plugins/org.jkiss.dbeaver.ext.bigquery/src/org/jkiss/dbeaver/ext/bigquery/model/data/BigQueryTrace.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/*
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.bigquery/src/org/jkiss/dbeaver/ext/bigquery/model/data/BigQueryTrace.java]: Move thus feature into com.dbeaver.db.bigquery

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.bigquery/src/org/jkiss/dbeaver/ext/bigquery/model/data/BigQueryTrace.java (hunk 2 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ext.bigquery/src/org/jkiss/dbeaver/ext/bigquery/model/data/BigQueryTrace.java]: Fixed

File: plugins/org.jkiss.dbeaver.ext.bigquery/src/org/jkiss/dbeaver/ext/bigquery/model/data/BigQueryTrace.java (hunk 3 above)
Comment (serjiokov) [reply plugins/org.jkiss.dbeaver.ext.bigquery/src/org/jkiss/dbeaver/ext/bigquery/model/data/BigQueryTrace.java]: Fixed

**Comment by serjiokov** [comment]
image

**Pull request merged**